cmd/collectors/ems/ems.go

```diff
@@ -574,128 +622,228 @@ func (e *Ems) HandleResults(result gjson.Result, prop map[string][]*emsProp) (ma
+					for k, v := range p.Labels {
+						instance.SetLabel(k, v)
+					}
 
-				//matches filtering
-				if len(p.Matches) == 0 {
-					isMatch = true
-				} else {
-					for _, match := range p.Matches {
-						if value := instance.GetLabel(match.Name); value != "" {
-							if value == match.value {
-								isMatch = true
-								break
+					//matches filtering
+					if len(p.Matches) == 0 {
+						isMatch = true
+					} else {
+						for _, match := range p.Matches {
+							if value := instance.GetLabel(match.Name); value != "" {
+								if value == match.value {
+									isMatch = true
+									break
+								}
+							} else {
+								//value not found
+								e.Logger.Warn().
+									Str("Instance key", instanceKey).
+									Str("name", match.Name).
+									Str("value", match.value).
+									Msg("label is not found")
 							}
-						} else {
-							//value not found
-							e.Logger.Warn().
-								Str("Instance key", instanceKey).
-								Str("name", match.Name).
-								Str("value", match.value).
-								Msg("label is not found")
 						}
 					}
-				}
-				if !isMatch {
-					instanceLabelCount = 0
-					continue
-				}
-
-				for _, metric := range p.Metrics {
-					metr, ok := mx.GetMetrics()[metric.Name]
-					if !ok {
-						if metr, err = mx.NewMetricFloat64(metric.Name); err != nil {
-							e.Logger.Error().Err(err).
-								Str("name", metric.Name).
-								Msg("NewMetricFloat64")
-						}
+					if !isMatch {
+						instanceLabelCount = 0
+						continue
 					}
-					if metric.Name == "events" {
-						if err = metr.SetValueFloat64(instance, 1); err != nil {
-							e.Logger.Error().Err(err).Str("key", metric.Name).Str("metric", metric.Label).
-								Msg("Unable to set float key on metric")
+
+					for _, metric := range p.Metrics {
+						metr, ok := mx.GetMetrics()[metric.Name]
+						if !ok {
+							if metr, err = mx.NewMetricFloat64(metric.Name); err != nil {
+								e.Logger.Error().Err(err).
+									Str("name", metric.Name).
+									Msg("NewMetricFloat64")
+								continue
+							}
+							metr.SetExportable(metric.Exportable)
 						}
-					} else {
-						// this code will not execute as ems only support events metric
-						f := instanceData.Get(metric.Name)
-						if f.Exists() {
-							if err = metr.SetValueFloat64(instance, f.Float()); err != nil {
+						if metric.Name == "events" {
+							if err = metr.SetValueFloat64(instance, 1); err != nil {
 								e.Logger.Error().Err(err).Str("key", metric.Name).Str("metric", metric.Label).
 									Msg("Unable to set float key on metric")
 							}
+						} else if metric.Name == "timestamp" {
+							if err = metr.SetValueFloat64(instance, float64(time.Now().UnixMicro())); err != nil {
+								e.Logger.Error().Err(err).Str("key", metric.Name).Str("metric", metric.Label).
+									Msg("Unable to set timestamp on metric")
+							}
+						} else {
+							// this code will not execute as ems only support events metric
```

Comment:
if this comment is correct, maybe there should be a logger.Warn() instead of setting a metric?

Reply:
done